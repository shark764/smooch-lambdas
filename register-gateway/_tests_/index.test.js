const axios = require('axios');

jest.mock('axios');

const event = {};

const mockGetSecretValue = jest.fn().mockImplementation(() => ({
  promise: () => ({
    SecretString: JSON.stringify({
      username: 'UserName',
      password: 'Password',
    }),
  }),
}));

axios.mockImplementation();

jest.mock('aws-sdk', () => ({
  SecretsManager: jest.fn().mockImplementation(() => ({
    getSecretValue: mockGetSecretValue,
  })),
}));

const { handler } = require('../index');

describe('register-gateway', () => {
  describe('Everything is successful', () => {
    it('returns when the code runs without any error', async () => {
      const result = handler(event);
      expect(result).toBeTruthy();
    });

    describe('Walkthrough', () => {
      beforeAll(async () => {
        jest.clearAllMocks();
        await handler(event);
      });

      it('passes in the correct arguments to secretsClient.getSecretValue() to get cx credentials', async () => {
        expect(mockGetSecretValue.mock.calls[0]).toMatchSnapshot();
      });

      it('passes in the correct arguments to axios', async () => {
        expect(axios.mock.calls).toMatchSnapshot();
      });
    });
  });

  it('throws an error when there is a problem trying to rettrieve cx credentials', async () => {
    try {
      mockGetSecretValue.mockRejectedValueOnce(new Error());
      await handler(event);
    } catch (error) {
      expect(Promise.reject(new Error('Error retrieving cx credentials'))).rejects.toThrowErrorMatchingSnapshot();
    }
  });

  it('throws an error when there is a problem callind endpoint to register gateway', async () => {
    try {
      axios.mockRejectedValueOnce(new Error());
      await handler(event);
    } catch (error) {
      expect(Promise.reject(new Error('Error calling API to register gateway'))).rejects.toThrowErrorMatchingSnapshot();
    }
  });
});
