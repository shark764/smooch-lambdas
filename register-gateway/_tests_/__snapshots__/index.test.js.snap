// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register-gateway Everything is successful Walkthrough passes in the correct arguments to axios 1`] = `
Array [
  Array [
    Object {
      "auth": Object {
        "password": "Password",
        "username": "UserName",
      },
      "data": Object {
        "actions": Array [
          Object {
            "name": "add-participant",
            "target": "arn:aws:sqs:AWS-REGION-NAME-us-east-1:460140541257:us-east-1-dev-smooch-action-add-participant",
          },
          Object {
            "name": "disconnect",
            "target": "arn:aws:sqs:AWS-REGION-NAME-us-east-1:460140541257:us-east-1-dev-smooch-action-disconnect",
          },
          Object {
            "name": "send-message",
            "target": "arn:aws:sqs:AWS-REGION-NAME-us-east-1:460140541257:us-east-1-dev-smooch-action-send-message",
          },
          Object {
            "name": "collect-message-response",
            "target": "arn:aws:sqs:AWS-REGION-NAME-us-east-1:460140541257:us-east-1-dev-smooch-action-collect-message-response",
          },
          Object {
            "name": "send-rich-message",
            "target": "arn:aws:sqs:AWS-REGION-NAME-us-east-1:460140541257:us-east-1-dev-smooch-action-send-rich-message",
          },
        ],
        "subscriptions": Array [],
        "type": "smooch",
        "url": "",
      },
      "method": "post",
      "url": "https://us-east-1-dev-edge.domain/v1/gateways",
    },
  ],
]
`;

exports[`register-gateway Everything is successful Walkthrough passes in the correct arguments to secretsClient.getSecretValue() to get cx credentials 1`] = `
Array [
  Object {
    "SecretId": "us-east-1-dev-smooch-cx",
  },
]
`;

exports[`register-gateway throws an error when there is a problem callind endpoint to register gateway 1`] = `"Error calling API to register gateway"`;

exports[`register-gateway throws an error when there is a problem trying to rettrieve cx credentials 1`] = `"Error retrieving cx credentials"`;
